docs/developers/backend/core/variables.rst

```diff
@@ -4,41 +4,42 @@
 Variables
 =========
 
-Submission Variables
-====================
+There are two models for variables, ``FormVariable`` and ``SubmissionValueVariable``. ``FormVariable`` are related to
+a form while ``SubmissionValueVariable`` are related to a submission.
 
-TODO: In #1705 will reorganise the info below
+Form Variables
+==============
 
-Static submission variables
----------------------------
+A ``FormVariable`` can have one of two sources:
 
-Static variables are not persisted to the database and are not sent to the frontend.
-They are evaluated in the backend and can be injected into the configuration or used
-during logic evaluation.
+* **Component**: the variable corresponds to a component in the form.
+* **User defined**:
+  the variable is added by the person designing the form and doesn't explicitly appear in the form. It
+  can however be used for calculations or in form fields. They can also be prefilled in the same way as form components.
 
-The flow of how static variables are used in the backend is:
+The API for the ``FormVariables`` has a bulk create / update endpoint. This means that a ``PUT`` call to this endpoint
+will delete all existing ``FormVariables`` related to a form and replace them with the data sent in the PUT call.
+If there are any existing completed submissions, the ``SubmissionValueVariables`` will not be deleted, but they will no
+longer be related to a ``FormVariable``. Any in progress submission will also have ``SubmissionValueVariables`` without
+a related ``FormVariable``. The result of this situation is that any data input by the user will not be saved.
```

Comment:
that last sentence confuses me - if the form definition changes while someone is filling it out, it'll still persist the values for the now-configured form variables/components, right? (assuming the form keys didn't change, and if they did, the user can go back a step and fix the input)

Reply:
Now that I stepped through it I actually think there is a bug :thinking: The current behaviour is what I described (if a user is filling out the form, but the definition is updated, then the data is not saved). It happens because no form variables are retrieved from this query:
https://github.com/open-formulieren/open-forms/blob/master/src/openforms/submissions/models/submission_value_variable.py#L100
(the `saved_submission_value_variables.values_list( "form_variable__id", flat=True)` contains a list with `None`). So I need to look into it :thinking:

Reply:
https://github.com/open-formulieren/open-forms/issues/1928